src/radix_connect/interaction_id.rs

```diff
@@ -0,0 +1,28 @@
+use crate::prelude::*;
+
+uniffi::custom_newtype!(WalletInteractionId, String);
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct WalletInteractionId(pub String);
+
+impl From<String> for WalletInteractionId {
+    fn from(value: String) -> Self {
+        Self(value)
+    }
+}
+
+impl From<&str> for WalletInteractionId {
+    fn from(value: &str) -> Self {
+        Self(value.to_string())
+    }
+}
+
+impl HasSampleValues for WalletInteractionId {
+    fn sample() -> Self {
+        Self("sample".to_string())
```

Comment:
and with the https://github.com/radixdlt/sargon/pull/98/files#r1574597240 you can drop the `to_string()` (in general, prefer using `to_owned()` when converting &str -> String, since it more accurately conveys the intent - into something owned.)